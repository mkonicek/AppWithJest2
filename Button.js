import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';

export default class Button extends Component {
  static propTypes = {
    text: React.PropTypes.string,
  };
  // TODO props
  render() {
    return (
      <TouchableOpacity style={styles.button}>
        <Text>
          {this.props.text}
        </Text>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  button: {
    borderWidth: 1,
    borderRadius: 4,
    borderColor: 'gray',
    padding: 10,
  }
});
