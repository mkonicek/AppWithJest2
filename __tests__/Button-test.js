import 'react-native';
import React from 'react';
import Button from '../Button';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Button', () => {

  it('renders correctly', () => {
    expect(renderer.create(
      <Button text={'hi'} />
    ).toJSON()).toMatchSnapshot()
  });

  // Advanced - See SUILink.react-test.js
  xit('expands help tooltip when tapped', () => {
    const component = ReactTestRenderer.create(
      <CreateScreen />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // Simulate button click - how?
    tree.button.onPress();
    // re-render
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});