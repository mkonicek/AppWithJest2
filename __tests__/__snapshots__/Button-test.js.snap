exports[`Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function bound touchableHandleResponderGrant]}
  onResponderMove={[Function bound touchableHandleResponderMove]}
  onResponderRelease={[Function bound touchableHandleResponderRelease]}
  onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
  onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
  onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
  style={
    Object {
      "borderColor": "gray",
      "borderRadius": 4,
      "borderWidth": 1,
      "opacity": 1,
      "padding": 10
    }
  }
  testID={undefined}>
  <Text>
    hi
  </Text>
</View>
`;
